.pre-commit-config.yaml

```diff
@@ -0,0 +1,48 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+  # isort should run before black as black sometimes tweaks the isort output
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+        args:
+          - --profile
+          - black
+          - -t
+          - py311
+          - src
+  # https://github.com/python/black#version-control-integration
+  - repo: https://github.com/psf/black
+    rev: 24.3.0
+    hooks:
+      - id: black
+        args:
+          - -t
+          - py311
+          - src
+        exclude: '.*\.ipynb$'
```

Comment:
Why are we applying black twice? Wouldn't the isort command be sufficient?

Reply:
yes, this is problem of `isort` that has kinda conflict with `black` and defining `profile` and utilizing `black` would be the solution to fix it. and also bringing `isort` before `black` is another way to not get conflict which i mentioned at top of `isort`